esmvalcore/preprocessor/_regrid.py

```diff
@@ -185,8 +189,45 @@ def _attempt_irregular_regridding(cube, scheme):
     return False
 
 
+def _add_regridded_aux_coords(cube, regridded_cube, target_grid):
+    """Get regridded :class:`iris.coords.AuxCoord` and corresponding dims."""
+    if regridded_cube.ndim != cube.ndim:
+        return
+    for coord in cube.coords(dim_coords=False):
+        if regridded_cube.coords(coord.name()):
+            continue
+        if coord.ndim < 2:
+            continue
+        if coord.bounds is not None:
+            logger.warning(
+                "Auxiliary coordinate '%s' cannot be regridded since it "
+                "contains bounds", coord.name())
+            continue
+        logger.debug(
+            "Regridding (bound-less) auxiliary coordinate '%s' which could "
+            "not be regridded by iris", coord.name())
+
+        # Broadcast to shape of whole cube
+        coord_dims = cube.coord_dims(coord)
+        points = iris.util.broadcast_to_shape(coord.core_points(),
+                                              cube.shape,
+                                              coord_dims)
+        aux_coord_cube = cube.copy(points)
+        aux_coord_cube = aux_coord_cube.regrid(
+            target_grid, Linear(extrapolation_mode='extrapolate'))
+
+        # Broadcast to original shape
+        slices = [0] * cube.ndim
+        for coord_dim in coord_dims:
+            slices[coord_dim] = slice(None)
+        aux_coord = coord.copy(aux_coord_cube[tuple(slices)].core_data())
+
+        # Add coordinate to regridded cube
+        regridded_cube.add_aux_coord(aux_coord, coord_dims)
```

Comment:
maybe make sure the `regridded_cube` doesn't already have the aux coord? It shouldn't, right?

Reply:
see comment above, l. 197